dbms/src/Functions/FunctionsDuration.cpp

```diff
@@ -97,6 +97,57 @@ void FunctionDurationSplit<Impl>::executeImpl(Block & block, const ColumnNumbers
             ErrorCodes::ILLEGAL_COLUMN);
 };
 
+template <typename Impl>
+DataTypePtr FunctionMyDurationToSec<Impl>::getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const
+{
+    if (!arguments[0].type->isMyTime())
```

Comment:
It seem type int, double and some strings can also work in MySQL.

```

mysql> SELECT TIME_TO_SEC(1234);
+-------------------+
| TIME_TO_SEC(1234) |
+-------------------+
|               754 |
+-------------------+
1 row in set (0.00 sec)

mysql> SELECT TIME_TO_SEC(12344);
+--------------------+
| TIME_TO_SEC(12344) |
+--------------------+
|               5024 |
+--------------------+
1 row in set (0.00 sec)

mysql> SELECT TIME_TO_SEC(12344.0);
+----------------------+
| TIME_TO_SEC(12344.0) |
+----------------------+
|                 5024 |
+----------------------+
1 row in set (0.00 sec)

mysql> SELECT TIME_TO_SEC(12344.1231231);
+----------------------------+
| TIME_TO_SEC(12344.1231231) |
+----------------------------+
|                       5024 |
+----------------------------+
1 row in set (0.00 sec)

mysql> SELECT TIME_TO_SEC("aasdf");
+----------------------+
| TIME_TO_SEC("aasdf") |
+----------------------+
|                 NULL |
+----------------------+
1 row in set, 1 warning (0.00 sec)

mysql> SELECT TIME_TO_SEC("123aasdf");
+-------------------------+
| TIME_TO_SEC("123aasdf") |
+-------------------------+
|                      83 |
+-------------------------+
1 row in set, 1 warning (0.00 sec)

```

Reply:
I don't think we should be worrying about this. See https://github.com/pingcap/tidb/blob/e50b9430fc650ce4629ead05186272ffedf014c8/expression/builtin_time.go#L6047, there is only one `evalInt` method for `builtinTimeToSecSig`, meaning that `time_to_sec(int/double/string)` is handled by appended implicit casts, which will be propagated to TiFlash as well. So we can safely assume the operands of `time_to_sec` is a `duration`.